CMakeLists.txt

```diff
@@ -111,6 +111,26 @@ add_library(mirage_runtime ${MIRAGE_SRCS})
 #set_target_properties(mirage_runtime
 #  PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
 
+#set RUST
+include(ExternalProject)
+set_directory_properties(PROPERTIES EP_PREFIX ${PROJECT_SOURCE_DIR}/deps/Rust)
+ExternalProject_Add(
+  rust_part
+  DOWNLOAD_COMMAND ""
+  CONFIGURE_COMMAND ""
+  BUILD_COMMAND cargo build --release
+  BINARY_DIR "${PROJECT_SOURCE_DIR}/rust_part"
+  INSTALL_COMMAND ""
+  LOG_BUILD ON)
+
+add_dependencies(mirage_runtime rust_part)
+
+target_link_libraries(mirage_runtime
```

Comment:
We determine the build type by searching the CMAKE_BUILD_TYPE string.

```
if ("${CMAKE_BUILD_TYPE}" STREQUAL "Debug")
  message("Build in Debug mode")
  set(CMAKE_CUDA_FLAGS "-O0 -g -Xcompiler=-fPIC ${CMAKE_CUDA_FLAGS}")
  set(CMAKE_CXX_FLAGS "-fopenmp -O0 -g -Wall -fPIC ${CMAKE_CXX_FLAGS}")
else()
  set(CMAKE_CUDA_FLAGS "-O2 -Xcompiler=-fPIC ${CMAKE_CUDA_FLAGS}")
  set(CMAKE_CXX_FLAGS "-fopenmp -O2 -Wall -fPIC ${CMAKE_CXX_FLAGS}")
endif()
```

It seems that the above rust command only builds `rust_part` in the release mode.